src/Particle/DTModes.h

```diff
@@ -0,0 +1,49 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2021 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_DTMODES_H
+#define QMCPLUSPLUS_DTMODES_H
+
+#include <cstdint>
+
+namespace qmcplusplus
+{
+enum class DTModes : uint_fast8_t
+{
+  ALL_OFF = 0x0,
+  /** whether full table needs to be ready at anytime or not
+   * Optimization can be implemented during forward PbyP move when the full table is not needed all the time.
+   * DT consumers should know if full table is needed or not and request via addTable.
+   */
+  NEED_FULL_TABLE_ANYTIME   = 0x1,
+  NO_NEED_TEMP_DATA_ON_HOST = 0x2
+};
+
```

Comment:
Since this is idiom and not straight forward some documentation is needed.
I don't think its very obvious that:
`DTModes dtmodes = static_cast<DTModes>(0x1 | 0x2)`
is ok.  I'm not sure if here is the place for the documentation or we want to have a section in the developer guide where we setout how we want to implement bitmasked options in the code going forward.

Reply:
Or add a unit test that shows the ways the enum class can be used. This would protect against someone replacing DTModes with another implementation that didn't work the same.

Reply:
No one should write ` DTModes dtmodes = static_cast<DTModes>(0x1 | 0x2)`
`DTModes dtmodes = DTModes::NEED_FULL_TABLE_ANYTIME | DTModes::NO_NEED_TEMP_DATA_ON_HOST` is the right way.

Reply:
Unit test added.